packages/angular/projects/angular-core/src/forms/checkbox/checkbox.directive.ts

```diff
@@ -14,56 +21,47 @@ import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
       provide: NG_VALUE_ACCESSOR,
       useExisting: forwardRef(() => CheckboxDirective),
       multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => CheckboxDirective),
+      multi: true
     }
   ]
 })
-export class CheckboxDirective implements ControlValueAccessor {
-  // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-explicit-any
-  onChange: (value: any) => void = () => {};
-  // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-explicit-any
-  onTouched: () => any = () => {};
+export class CheckboxDirective extends FormControlElementDirective {
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private _value: any;
+  #initialValue: string | undefined;
```

Comment:
Should this not be `#initialValue?: string;`

Reply:
The value for the checkbox web component can be string or undefined, that's why I have it here string or undefined, to have the validation working properly.